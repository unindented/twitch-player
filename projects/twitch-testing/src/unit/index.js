import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import React from "react";
import { render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>{node}</ThemeProvider>,
    ...options
  );
};

const fireResizeEvent = (el, { left = 0, top = 0, width = 0, height = 0 }) => {
  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const evt = document.createEvent("Event");
  evt.initEvent("resize", true, true);
  el.dispatchEvent(evt);
};

export * from "react-testing-library";

export { customRender as render, fireResizeEvent };
