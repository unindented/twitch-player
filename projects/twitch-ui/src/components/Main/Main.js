import PropTypes from "prop-types";
import React from "react";
import { StyleSheet, View } from "react-native";
import { useTheme } from "../../hooks";

const Main = ({ children, testID = "main" }) => {
  const theme = useTheme();

  return (
    <View
      accessibilityRole="main"
      style={[
        styles.root,
        {
          backgroundColor: theme.colors.bodyBackground,
          padding: theme.layout.gapBig,
        },
      ]}
      testID={testID}
    >
      {children}
    </View>
  );
};

Main.propTypes = {
  children: PropTypes.node.isRequired,
  testID: PropTypes.string,
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
  },
});

export default Main;
