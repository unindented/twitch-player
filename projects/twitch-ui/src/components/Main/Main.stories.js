import { storiesOf } from "@storybook/react";
import React from "react";
import Spinner from "../Spinner";
import Main from ".";

storiesOf("Components|Main", module).add("default", () => (
  <Main>
    <Spinner />
  </Main>
));
