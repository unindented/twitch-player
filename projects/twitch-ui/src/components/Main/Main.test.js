import { render } from "@twitch-player/testing/dist/unit";
import React from "react";
import { Text } from "react-native";
import Main from "./Main";

describe("Main", () => {
  let instance;

  beforeEach(() => {
    instance = render(
      <Main>
        <Text>OHAI</Text>
      </Main>
    );
  });

  it("renders a main element", () => {
    expect(instance.getByTestId("main")).toMatchSnapshot();
  });
});
