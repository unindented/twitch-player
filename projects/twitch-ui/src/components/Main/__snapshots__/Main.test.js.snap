// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders a main element 1`] = `
<main
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexGrow-16y2uox rn-flexShrink-1wbh5a2 rn-flexBasis-1ro0kt6 rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-position-bnwqim rn-zIndex-1lgpqti"
  data-testid="main"
  role="main"
  style="background-color: rgb(250, 249, 250); padding: 20px 20px 20px 20px;"
>
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
  >
    OHAI
  </div>
</main>
`;
